<?xml version="1.0"?>
<robot name="model_with_non_zero_offset_joint">
<!--
Implements a 1-DOF robot with a joint that's located at (x = 0, y = 0, z = 1)
relative to its parent body's coordinate frame.
-->
  <link name="link1">
    <inertial>
      <mass value="1"/>
      <inertia ixx="0.0325" ixy="0.0" ixz="0.0" iyy="0.0325" iyz="0.0" izz="0.005"/>
      <origin xyz="0 0 0.3"/>
    </inertial>
    <visual>
      <origin xyz="0 0 0.3" rpy="0 0 0"/>
      <geometry>
        <cylinder length="0.6" radius="0.1"/>
      </geometry>
     <material name="Cyan">
       <color rgba="0 1.0 1.0 1.0"/>
     </material>
    </visual>
    <collision>
      <geometry>
        <cylinder length="0.6" radius="0.1"/>
      </geometry>
      <origin xyz="0 0 0.3"/>
    </collision>
  </link>

  <link name="link2">
    <inertial>
      <mass value="1"/>
      <inertia ixx="0.0325" ixy="0.0" ixz="0.0" iyy="0.0325" iyz="0.0" izz="0.005"/>
      <origin xyz="0 0 0.3"/>
    </inertial>
    <visual>
      <origin xyz="0 0 0.3" rpy="0 0 0"/>
      <geometry>
        <cylinder length="0.6" radius="0.1"/>
      </geometry>
     <material name="Cyan">
       <color rgba="0 1.0 1.0 1.0"/>
     </material>
    </visual>
    <collision>
      <geometry>
        <cylinder length="0.6" radius="0.1"/>
      </geometry>
      <origin xyz="0 0 0.3"/>
    </collision>
  </link>

  <joint name="joint1" type="revolute">
    <parent link="link1"/>
    <child link="link2"/>
    <origin xyz="0 0 1" rpy="0 0 0"/>
    <limit effort="30" velocity="1.0" lower="-1.5708" upper="1.5708" />
  </joint>

</robot>